src/wallet/wallet.cpp

```diff
@@ -2480,6 +2480,17 @@ void CWallet::UpdatedTransaction(const uint256 &hashTx)
     }
 }
 
+void CWallet::GetScriptForMining(CScript &script)
+{
+    CReserveKey reservekey(this);
+    reservekey.KeepKey();
```

Comment:
IIRC there was a discussion about this on IRC.
But right, i also have a bad feeling with this. Every mining thread reserves and keeps a key straight off.

I think i'm going invest there some time and build a proper way of only keeping keys which where used when a block was created.